Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -163,6 +168,19 @@ void Q1DWeighted::exec() {
 
   const auto &spectrumInfo = inputWS->spectrumInfo();
 
+  // Create workspace group that holds output workspaces
+  std::vector<MatrixWorkspace_sptr> tofWorkspaces;
+  for (int iTOF = 0; iTOF < xLength - 1; iTOF++) {
+    MatrixWorkspace_sptr tof_ws =
+        WorkspaceFactory::Instance().create(inputWS, 1, sizeOut, sizeOut - 1);
+    tof_ws->getAxis(0)->unit() =
+        UnitFactory::Instance().create("MomentumTransfer");
+    tof_ws->setYUnitLabel("1/cm");
+    tof_ws->setDistribution(true);
+    tof_ws->setBinEdges(0, XOut);
+    tofWorkspaces.push_back(tof_ws);
```

Comment:
I think it would be better to change this line to `tofWorkspaces.push_back(std::move(tof_ws));` Also, is `xLength - 1` large enough to justify reserving space in `tofWorkspaces`?